File: server/php/src/Application/Domain/Payments/PaymentDetails/HostedCheckoutSpecificOutput.php
<?php

namespace OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails;

final class HostedCheckoutSpecificOutput
{
    public ?string $hostedCheckoutId = null;
    public ?string $variant = null;

    public function toArray(): array
    {
        return [
            'hostedCheckoutId' => $this->hostedCheckoutId,
            'variant' => $this->variant,
        ];
    }
}

File: server/ruby/app/business/domain/payments/payment_details/hosted_checkout_specific_output.rb
module Business
  module Domain
    module Payments
      module PaymentDetails
        class HostedCheckoutSpecificOutput
          attr_accessor :hosted_checkout_id, :variant

          def initialize(hosted_checkout_id: nil, variant: nil)
            @hosted_checkout_id = hosted_checkout_id
            @variant = variant
          end

          def to_h
            {
              'hostedCheckoutId' => @hosted_checkout_id,
              'variant' => @variant
            }
          end
        end
      end
    end
  end
end